galasa-ui/src/app/mysettings/page.tsx

```diff
@@ -0,0 +1,156 @@
+import TokenCard from "@/components/TokenCard";
+import ErrorPage from "../error/page";
+import TokenRequestModal from "@/components/TokenRequestModal";
+import TokenDeleteModal from "@/components/TokenDeleteModal";
+import TokenResponseModal from "@/components/TokenResponseModal";
+
+export default function MySettingsPage() {
+
+    const [isLoading, setIsLoading] = useState(false)
+    const [isError, setIsError] = useState(false)
+    const [selectedTokens, setSelectedTokens] = useState<Set<string>>(new Set());
+    const [tokens, setTokens] = useState<Set<Token>>(new Set())
+    const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false)
+
+    interface Token {
+        tokenId: string;
+        description: string;
+        creationTime: string;
+        owner: {
+            loginId: string;
+        };
+    }
+
+    const handleSelectTokensForDeletion = (tokenId: any) => {
+
+        if (selectedTokens.has(tokenId)) {
+            selectedTokens.delete(tokenId);
+
+            setSelectedTokens((prevSelectedTokens) => {
+                const newSet = new Set(prevSelectedTokens);
+                newSet.delete(tokenId);
+                return newSet;
+            });
+        }
+        else {
+            setSelectedTokens((prevSelectedTokens) => {
+                const newSet = new Set(prevSelectedTokens);
+                newSet.add(tokenId);
+                return newSet;
+            });
+
+        }
+    }
+
+    const handleFetchAccessTokens = async () => {
+
+        setIsLoading(true)
+
+        try {
+            const response = await fetch('/auth/tokens', { method: 'GET' })
+
+            if (response.ok) {
+
+                const data = await response.json();
+
+                setTokens(data)
+
+            }
+
+        } catch (err) {
+            setIsError(true)
+        }
+        finally {
+            setIsLoading(false)
+        }
+
+    }
+
+    const handleUpdateDeleteModalState = () => {
+        setIsDeleteModalOpen(false)
+    }
+
+    const handleUpdateTokens = (token : any) => {
+        setTokens((prevTokens) => {
+            const newTokens = new Set(prevTokens);
+            newTokens.delete(token);
+            return newTokens;
+        });
+
+        setSelectedTokens((prevSelectedTokens) => {
+            const newSelectedTokens = new Set(prevSelectedTokens);
+            newSelectedTokens.delete(token.tokenId);
+            return newSelectedTokens;
+        });
+
+        setIsDeleteModalOpen(false)
+    }
+
+    useEffect(() => {
+
+        handleFetchAccessTokens()
+
+    }, [])
+
+    if (isLoading) {
+        return <Loading />
+    }
+
+    if (isError) {
+        return (
+            <ErrorPage />
+        )
+    }
+
+    return (
+        <div id="content" className={styles.content}>
+            <PageTile title={"My Settings"} />
+
+            <div className={styles.tokenContainer}>
+                <h4 style={{ margin: "1rem 0" }}>Access Tokens</h4>
```

Comment:
Should this style be in a stylesheet instead of inlined to keep the separation between styling and markup?